crates/rooch-types/src/address.rs

```diff
@@ -515,32 +514,48 @@ impl BitcoinAddress {
         Self { bytes }
     }
 
-    pub fn new_p2pkh(pubkey_hash: &bitcoin::PubkeyHash) -> Self {
+    pub fn get_pubkey_address_prefix(network: u8) -> u8 {
+        if network::Network::NetworkBitcoin.to_num() == network {
+            bitcoin::constants::PUBKEY_ADDRESS_PREFIX_MAIN
+        } else {
+            bitcoin::constants::PUBKEY_ADDRESS_PREFIX_TEST
+        }
+    }
+
+    pub fn get_script_address_prefix(network: u8) -> u8 {
+        if network::Network::NetworkBitcoin.to_num() == network {
+            bitcoin::constants::SCRIPT_ADDRESS_PREFIX_MAIN
+        } else {
+            bitcoin::constants::SCRIPT_ADDRESS_PREFIX_TEST
+        }
+    }
+
+    pub fn new_p2pkh(pubkey_hash: &bitcoin::PubkeyHash, network: u8) -> Self {
         let mut bytes = [0; 22];
         bytes[0] = BitcoinAddressPayloadType::PubkeyHash.to_num();
-        //we always use mainnet prefix, do not distinguish testnet and mainnet in Move contract
-        bytes[1] = bitcoin::constants::PUBKEY_ADDRESS_PREFIX_MAIN;
+        bytes[1] = Self::get_pubkey_address_prefix(network);
         bytes[2..].copy_from_slice(&pubkey_hash[..]);
         Self {
             bytes: bytes.to_vec(),
         }
     }
 
-    pub fn new_p2sh(script_hash: &bitcoin::ScriptHash) -> Self {
+    pub fn new_p2sh(script_hash: &bitcoin::ScriptHash, network: u8) -> Self {
```

Comment:
Should Rooch BitcoinAddress care about the Bitcoin network?
Can we ignore the network in the Move contracts and only add the prefix when JSON RPC to string?

Reply:
done